velox/expression/FunctionCallToSpecialForm.cpp

```diff
@@ -31,6 +31,16 @@ TypePtr resolveTypeForSpecialForm(
   return specialForm->resolveType(argTypes);
 }
 
+TypePtr resolveTypeForSpecialForm(
+    const std::string& functionName,
+    const std::vector<std::shared_ptr<const core::ITypedExpr>>& inputs) {
```

Comment:
I'm not sure if we could add the function or change the old function `resolveTypeForSpecialForm` input type from `const std::vector<TypePtr>& argTypes` to current `const std::vector<std::shared_ptr<const core::ITypedExpr>>& inputs`, this will affect all the SpecialForm.

Reply:
I have a feeling that in the future, we might need both input types and inputs.
Can we extend the old function to include `const std::vector<std::shared_ptr<const core::ITypedExpr>>& inputs` as well? May be default to an empty vector if that is easier?

Reply:
But `inputTypes` are actually derived from the `inputs`. So I guess passing inputs everywhere makes sense.
@kevinwilfong any thoughts here?